tools/clang/test/CodeGenHLSL/SimpleHs10.hlsl

```diff
@@ -0,0 +1,98 @@
+// RUN: %dxc -E main -T hs_6_1  %s 2>&1 | StdErrCheck %s
+
+// Same as SimpleHS11.hlsl, except that we only verify StdErr for the warning
+// message.
+
+// CHECK: Multiple overloads of patchconstantfunc HSPerPatchFunc.
+// CHECK: 88:16: note: This overload was selected
+
+//--------------------------------------------------------------------------------------
+// SimpleTessellation.hlsl
+//
+// Advanced Technology Group (ATG)
+// Copyright (C) Microsoft Corporation. All rights reserved.
+//--------------------------------------------------------------------------------------
+
+
+struct PSSceneIn
+{
+    float4 pos     : SV_Position;
+    float2 tex     : TEXCOORD0;
+    float3 norm    : NORMAL;
+    uint   RTIndex : SV_RenderTargetArrayIndex;
+};
+
+
+//////////////////////////////////////////////////////////////////////////////////////////
+// Simple forwarding Tessellation shaders
+
+struct HSPerVertexData
+{
+    // This is just the original vertex verbatim. In many real life cases this would be a
+    // control point instead
+    PSSceneIn v;
+};
+
+struct HSPerPatchData
+{
+    // We at least have to specify tess factors per patch
+    // As we're tesselating triangles, there will be 4 tess factors
+    // In real life case this might contain face normal, for example
+	float	edges[3] : SV_TessFactor;
+	float	inside   : SV_InsideTessFactor;
+};
+
+// This function has the same name as the patch constant function, but
+// its signature prevents it from being a candidate.
+float4 HSPerPatchFunc()
+{
+    return 1.8;
+}
+
+// The compiler should *not* select this overload as the patch constant
```

Comment:
Not clear about the reason the compiler should *not* select this overload here.  The comments seem to imply that it's because the InputPatch dimension doesn't match the hull shader, but that's not the actual reason.  The reason is because this is not the last patch constant function candidate by the same name in the file.  A patch constant function candidate is a function that has an output with an SV_TessFactor semantic.  We should probably make it clear somewhere that this is the behavior.  Also, it matches the old compiler in the way that matters (for valid cases).

Reply:
Thanks for the clarification. I modified the comments. Please let me know if they are more clear now.